listeners/flow.py

```diff
@@ -0,0 +1,147 @@
+#
+#  Copyright 2019     Qentinel
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Flow is a graphml parser for RobotFlow.
+"""
+from pathlib import Path
+import xml.etree.ElementTree as ET
+from robot.libraries.BuiltIn import BuiltIn
+
+
+class Flow():
+    """ Custom class for flow control """
+
+    def __init__(self, graph, initial=None, verbose=False):
+        self.verbose = verbose
+        self.graph = graph
+        if initial:
+            self.initial = initial
+        else:
+            self.initial = self.parse_initial()
+        self._state = None
+        self.running = False
+
+    @classmethod
+    def from_graphml(cls, script_file, verbose=False):
+        """Parse grapml.
+        """
+        graph_file = script_file.split(".")[0] + ".graphml"
+        if verbose:
+            print("Script file: {}".format(script_file))
+            print("GraphML file: {}".format(graph_file))
+        if not Path(graph_file).exists():
+            raise FileNotFoundError("File not found: {}".format(graph_file))
+
+        tree = ET.parse(str(graph_file))
```

Comment:
As file is no longer needed, it could be closed right after this (or just use with to open it, so Python will close it right away. Not a huge issue right now as there's only one file opened and Python will close it anyway. Though quite easy to fix.